pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -78,16 +85,16 @@ type InTreeToCSITranslator interface {
 // 2. Once all the assume operations are done in d), the scheduler processes the next Pod in the scheduler queue
 //    while the actual binding operation occurs in the background.
 type SchedulerVolumeBinder interface {
-	// FindPodVolumes checks if all of a Pod's PVCs can be satisfied by the node.
+	// FindPodVolumes checks if all of a Pod's volumes can be satisfied by the node.
 	//
 	// If a PVC is bound, it checks if the PV's NodeAffinity matches the Node.
 	// Otherwise, it tries to find an available PV to bind to the PVC.
 	//
-	// It returns true if all of the Pod's PVCs have matching PVs or can be dynamic provisioned,
-	// and returns true if bound volumes satisfy the PV NodeAffinity.
+	// It returns an error when something went wrong or a list of reasons why the node is
+	// (currently) not usable for the pod.
 	//
 	// This function is called by the volume binding scheduler predicate and can be called in parallel
-	FindPodVolumes(pod *v1.Pod, node *v1.Node) (unboundVolumesSatisified, boundVolumesSatisfied bool, err error)
+	FindPodVolumes(pod *v1.Pod, node *v1.Node) (reasons []string, err error)
```

Comment:
should we make the reasons typed so that we can easily search and find the reasons?

Reply:
The reason might also be the result of a `fmt.Sprintf`, it doesn't necessarily have to be a constant. Such a string could of course also be typecast - okay, let's introduce a type.

Reply:
Done. This had implications throughout the code. I've kept it in separate commit, so it is possible to compare the version with and without special types in the PR.

@msau42 what do you prefer?